File: TIC10002/src/main/java/uff/ic/lleme/tic10002/trabalhos/s20181/Lucas_e_Erick/src/br/uff/ed/main/Assunto.java
package br.uff.ed.main;

@SuppressWarnings("rawtypes")
public class Assunto implements Comparable {
private TipoAssunto tipoAssunto;
private String descricao;
private String providencia;
private int duracao;

public Assunto(TipoAssunto tipoAssunto, String descricao, String providencia, int duracao) {
	this.tipoAssunto = tipoAssunto;
	this.descricao = descricao;
	this.providencia = providencia;
	this.duracao = duracao;
}

public TipoAssunto getTipoAssunto() {
	return tipoAssunto;
}

public void setTipoAssunto(TipoAssunto tipoAssunto) {
	this.tipoAssunto = tipoAssunto;
}

public String getDescricao() {
	return descricao;
}

public void setDescricao(String descricao) {
	this.descricao = descricao;
}

public String getProvidencia() {
	return providencia;
}

public void setProvidencia(String providencia) {
	this.providencia = providencia;
}

public int getDuracao() {
	return duracao;
}

public void setDuracao(int duracao) {
	this.duracao = duracao;
}
@Override
public int compareTo(Object assunto){
	return 0;
}
}

File: TIC10002/src/main/java/uff/ic/lleme/tic10002/trabalhos/s20181/GabrielDuarte_CintiaCosta/Assunto.cpp
/*

	Gabriel Duarte
	Cintia Costa

*/

#ifndef ASSUNTO
#define ASSUNTO 1
#include <bits/stdc++.h>
#include "TipoAssunto.cpp"
using namespace std;

class Assunto{
private:
	TipoAssunto tipoAssunto;
	string descricao, providencias;
	int duracaoAtendimento;
public:
	Assunto(){}
	Assunto(TipoAssunto tp, string d){
		tipoAssunto = tp;
		descricao = d;
	}
	void setTipoAssunto(TipoAssunto t){ tipoAssunto = t; }
	void setDescricao(string d){ descricao = d; }
	void setProvidencias(string p){ providencias = p; }
	void setDuracao(int d){ duracaoAtendimento = d; }
	TipoAssunto getTipoAssunto(){ return tipoAssunto; }
	string getDescricao(){	return descricao; }
	string getProvidencias(){ return providencias; }
	int getDuracao(){ return duracaoAtendimento; }

	void imprimirAssunto(){
		printf("Tipo assunto: %s - descricao: %s\n", tipoAssunto.getTitulo().c_str(), descricao.c_str());
	}

};	

#endif